containenv/containenv/cli/commands/test/test_init.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+
+# Copyright 2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+import os
+
+import pytest
+
+from containenv.exceptions import ProjectAlreadyInitialized,\
+                                  ProjectDirectoryDoesNotExist,\
+                                  ProjectContainsNoRegisteredNodes
+from containenv.containenv.cli.commands.init import Command as Init
+
+logger = logging.getLogger(__name__)
+
+
+# tests pre self.make_containenv_dir
+def test__check_project_existence_no_dir(testdirectory):
+    '''The argument does not a even correspond to a directory.'''
+    NONEXTANTDIR = os.path.join(testdirectory, 'NONE')
+    EXPECTED_ERROR = ('init must be called on a project directory but the'
+                      ' argument it received was: {}\nwhich does not map to a'
+                      ' path to an existing directory.'.format(NONEXTANTDIR))
+    init_command = Init(NONEXTANTDIR)
+    init_command.make_path()
+    with pytest.raises(ProjectDirectoryDoesNotExist) as PDDNEEIO:
+        init_command._check_project_existence()
+    assert PDDNEEIO.value.args[0] == EXPECTED_ERROR
+
+
+def test__check_project_existence_dir(testdirectory):
+    '''The argument does correspond to a directory.'''
+    init_command = Init(testdirectory)
+    init_command.make_path()
+    assert init_command._check_project_existence() is None
+
+
+def test__check_initialization_state_uninitialized(testdirectory):
+    '''The argument does not _already_ possess a ".containenv".'''
+    init_command = Init(testdirectory)
+    init_command.make_path()
+    assert init_command.check_initialization_state() is None
+
+
+# tests post self.make_containenv_dir
+@pytest.fixture
+def testdirskeleton(testdirectory):
+    SUBDIRPATH = os.path.join(testdirectory, 'SUB')
+    os.mkdir(SUBDIRPATH)
+    init_command = Init(testdirectory)
+    init_command.make_path()
+    init_command.make_containenv_dir()
+    return testdirectory, init_command.containenvdir, SUBDIRPATH, init_command
+
+
+def test__check_initialization_state_already_initialized(testdirskeleton):
+    '''The argument _already_ (incorrectly) possesses a ".containenv".'''
+    testdirectory, CONTAINENVPATH, SUBDIRPATH, init_command = testdirskeleton
+    EXPECTED_ERROR = ('init can only be called once per project, but the {}\n'
```

Comment:
OK, can we leave the fragiles in, and use best practice going forward..  (e.g. change to value checks *WHEN* the messages change?)